jobs/data_quality.py

```diff
@@ -0,0 +1,203 @@
+import re
+from nautobot.extras.jobs import Job, StringVar, MultiObjectVar
+from nautobot.dcim.models import Device, Site, DeviceRole, DeviceType
+from nautobot.circuits.models import Circuit
+
+
+name = "Data Quality"
+
+
+def normalize(query_set):
+    """Returns a list of names for the log entry
+    
+    Args:
+        query_set: QuerySet object
+    """
+    list_of_names = []
+    for element in query_set:
+        if hasattr(element, "name"):
+            list_of_names.append(element.name)
+        else:
+            list_of_names.append(element.slug)
+    return ', '.join(list_of_names)
+
+
+def filter_devices(data, log):
+    """Returns a list of devices per filter parameters
+    
+    Args:
+        data: A dictionary from the job input
+        log: The log instance for logs
+    """
+
+    devices = Device.objects.all()
+
+    site = data["site"]
+    if site:
+        log(f"Filter sites: {normalize(site)}")
+        # *__in enables to pass the query set as a parameter
+        devices = devices.filter(site__in=site)
+
+    device_role = data["device_role"]
+    if device_role:
+        log(f"Filter device roles: {normalize(device_role)}")
+        devices = devices.filter(device_role__in=device_role)
+
+    device_type = data["device_type"]
+    if device_type:
+        log(f"Filter device types: {normalize(device_type)}")
+        devices = devices.filter(device_type__in=device_type)
+
+    return devices
+
+
+class FormData:
+
+    site = MultiObjectVar(
+        model = Site,
+        required = False,
+    )
+    device_role = MultiObjectVar(
+        model = DeviceRole,
+        required = False
+    )
+    device_type = MultiObjectVar(
+        model = DeviceType,
+        required = False
+    )
+
+
+class VerifyHostnames(Job):
+    """Demo job that verifies device hostnames match corporate standards."""
+
+    class Meta:
+        """Meta class for VerifyHostnames"""
+
+        name = "Verify Hostnames"
+        description = "Verify device hostnames match corporate standards"
```

Comment:
Given that these jobs do not modify the database, they should each declare `read_only = True` in their Meta classes. (https://nautobot.readthedocs.io/en/stable/additional-features/jobs/#read_only)